### PR TITLE
Update to scala 2.12

### PR DESCRIPTION
**Pull request by FlorinAcsinte**
Update the latest 1.X elastic4s version to work with scala 2.12

### CODE REVIEW

**Comment by sksamuel** [comment]
Can you update travis to run against 2.12 as part of this PR.

**Comment by FlorinAcsinte** [comment]
@sksamuel Updated travis to also run against scala 2.12

**Comment by sksamuel** [comment]
Great thanks. You want a release of this?

**Comment by FlorinAcsinte** [comment]
If you could, that would be great. Thanks!

**Comment by FlorinAcsinte** [comment]
@sksamuel Hi. Do you know when you can create a new release for this branch? Also, could you please publish it to the maven repo so that it can be easily used with sbt?

Thanks!

**Comment by FlorinAcsinte** [comment]
@sksamuel Hi. Sorry to be bothering you again with this, but do you know when this release (1.7.7) will be available on mvnrepository ? Is there anything I can do to help with doing this?

**Comment by FlorinAcsinte** [comment]
I see the release on the github releases page (which contains the source code) but on mvnrepository it is not present. The latest 1.7 release over there is 1.7.6 (link to all releases: https://mvnrepository.com/artifact/com.sksamuel.elastic4s/elastic4s-core). Looking on search.maven.com also doesn't show the 1.7.7 version (link: https://search.maven.org/search?q=elastic4s)

**Comment by sksamuel** [comment]
If you look in the actual repo, you can see 1.7.7 has been released:
http://repo1.maven.org/maven2/com/sksamuel/elastic4s/elastic4s-core_2.12/
And it does show up on search.
https://search.maven.org/search?q=g:com.sksamuel.elastic4s%20AND%20a:elastic4s-core_2.12&core=gav

mvnrepository is always last to update

**Comment by FlorinAcsinte** [comment]
Didn't knew that. Thanks for the info and for the links!
Cheers!

File: project/Build.scala [hunk 1]
```diff
@@ -6,13 +6,13 @@ object Build extends Build {
...
-  val ScalatestVersion = "2.2.5"
+  val ScalaVersion = "2.12.6"
+  val ScalatestVersion = "3.0.5"
   val MockitoVersion = "1.9.5"
-  val JacksonVersion = "2.6.1"
-  val Slf4jVersion = "1.7.12"
-  val ScalaLoggingVersion = "2.1.2"
-  val ElasticsearchVersion = "1.7.2"
+  val JacksonVersion = "2.9.6"
+  val Slf4jVersion = "1.8.0-beta2"
```
Comment (cloudjohnson) [inline project/Build.scala]: @FlorinAcsinte, can you elaborate on why you updated to the beta version of slf4j? We would like to use the 1.7.x branch with scala 2.12, but this specific dependency upgrade blocks us.

**Pull request merged**